mantidimaging/gui/windows/spectrum_viewer/presenter.py

```diff
@@ -118,15 +118,25 @@ def handle_range_slide_moved(self, tof_range) -> None:
         self.model.tof_range = tof_range
         self.view.set_image(self.model.get_averaged_image(), autoLevels=False)
 
+    def check_for_changed_rois(self) -> Optional[str]:
+        """
+        Return a list of rois which have changed in size and or position
+        """
+        for name in self.model.get_list_of_roi_names():
+            roi = self.view.spectrum.get_roi(name)
+            if roi != self.model.get_roi(name):
+                return name
+        return None
+
     def handle_roi_moved(self) -> None:
         """
         Handle changes to any ROI position and size.
         """
-        roi_names = self.model.get_list_of_roi_names()
-        for name in roi_names:
-            roi = self.view.spectrum.get_roi(name)
-            self.model.set_roi(name, roi)
-            self.view.set_spectrum(name, self.model.get_spectrum(name, self.spectrum_mode))
+        roi_name = self.check_for_changed_rois()
+        if roi_name is not None and roi_name in self.view.spectrum.roi_dict.keys():
+            roi = self.view.spectrum.get_roi(roi_name)
+            self.model.set_roi(roi_name, roi)
+            self.view.set_spectrum(roi_name, self.model.get_spectrum(roi_name, self.spectrum_mode))
 
```

Comment:
Would it simpler to keep the old loop and add a guard
```
if roi == self.model.get_roi(name):
    continue
```
instead of a new function and testing on None.